### PR TITLE
framework/st_things : improve things ping.

### PR DESCRIPTION
**Pull request by Hyunjun85**
It has shown an error when reset request receives from client.
The error cause is racing between ping, ca and session keep alive threads.

1. The ping apply mutex logic to common used variable.
2. Ping thread will terminate when tcp session disconnected.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [24ab5083a10cf4b068e8e5c5eb3b86ffb67e0cff] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [24ab5083a10cf4b068e8e5c5eb3b86ffb67e0cff] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [24ab5083a10cf4b068e8e5c5eb3b86ffb67e0cff] -

File: framework/src/st_things/things_stack/utils/things_ping.c [hunk 1]
```diff
@@ -121,21 +128,28 @@ bool things_ping_terminate(void)
...
 		THINGS_LOG_D(TAG, "things_ping_s terminate.");
+
 		while ((ping = (things_ping_s *) list->pop(list)) != NULL) {
 			THINGS_LOG_D(TAG, "Terminate ping.(%s)", ping->addr);
 			if (cas_mask(ping, PING_ST_STARTTHREAD, true, PING_ST_INIT) == true) {
 				things_ping_destroy_thread(ping);
 			}
 			terminate_things_ping_s(ping);
+
+			THINGS_LOG_D(TAG, ".");
```
Comment (JeongJunSik) [inline framework/src/st_things/things_stack/utils/things_ping.c]: What is this log mean?

File: framework/src/st_things/things_stack/utils/things_ping.c [hunk 2]
```diff
@@ -152,6 +166,7 @@ bool things_ping_set_mask(const char *remote_addr, uint16_t port, ping_state_e s
 		return false;
 	}
 
+	pthread_mutex_lock(&mutex_ping_list);
 	if (list == NULL) {
 		THINGS_LOG_V(TAG, "OICPing Module is not initialized.");
 		return false;
```
Comment (JeongJunSik) [inline framework/src/st_things/things_stack/utils/things_ping.c]: you need to unlock here.

Comment (Hyunjun85) [reply framework/src/st_things/things_stack/utils/things_ping.c]: The mutex will be unlocked at line no. 187.

File: framework/src/st_things/things_stack/utils/things_ping.c (hunk 1 above)
Comment (Hyunjun85) [reply framework/src/st_things/things_stack/utils/things_ping.c]: Oops, It has removed.

**Comment by Hyunjun85** [comment]
@JeongJunSik please review again. I have applied your opinions.

**Comment by seinfra** [comment]
Target : [7364b19ba2dbfe30d44d45de156d0aa9b5d20e99] -

**Comment by seinfra** [comment]
Target : [7364b19ba2dbfe30d44d45de156d0aa9b5d20e99] -

**Comment by seinfra** [comment]
Target : [cbe549751e52778d8dce3f3f85d57d5505cee6c0] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cbe549751e52778d8dce3f3f85d57d5505cee6c0] -

**Comment by seinfra** [comment]
Target : [cbe549751e52778d8dce3f3f85d57d5505cee6c0] - Code Rule Check (C++) OK.

**Pull request merged**